metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/api/SraObjectsImporterThroughApi.java

```diff
@@ -158,4 +163,24 @@ protected Analysis extractStudyFromAnalysis(AnalysisType analysisType, Analysis
 
         return analysisAccessions;
     }
+
+    @Override
+    @Transactional(rollbackFor = Exception.class)
+    public Study importStudy(String accession) throws Exception {
```

Comment:
Is there a reason to duplicate code from `ObjectsImporter` and `SraXmlRetrieverThroughApi` here? It seems like it would be sufficient to just add a `@Retryable` annotation to `SraXmlRetrieverThroughApi.getXml()`

Reply:
Spring will not be able to resolve. Below error will be thrown
"Bean named 'sraXmlRetrieverThroughApi' is expected to be of type 'uk.ac.ebi.ampt2d.metadata.importer.api.SraXmlRetrieverThroughApi' but was actually of type 'com.sun.proxy.$Proxy137'"

Reply:
This appears to be because we're relying on a specific implementation (`SraXmlRetrieverThroughApi`) rather than the interface (`SraXmlRetrieverByAccession`). It looks like this can can be fixed by relying on an interface (which also makes more sense). See here, for example: https://stackoverflow.com/questions/14509142/beannotofrequiredtypeexception-due-to-autowired-fields

In either case, code duplication is not a good option

Reply:
Thanks, ObjectImporter->importStudy() has sraXmlRetrieverByAccession.getXml(). Did i miss anything

Reply:
For example, see `MetadataImporterMainApplicationConfiguration`, where the autowiring actually happens. This is where we autowire by specific class (`SraXmlRetrieverThroughApi sraXmlRetrieverThroughApi`) rather than by an interface.

I think something like this should work: https://stackoverflow.com/a/31138957

Reply:
I. e. make it like `SraXmlRetrieverByAccession sraXmlRetrieverThroughApi`